src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -39,10 +39,15 @@ public DiagnosticState GetState(StateType stateType)
                 return _state[(int)stateType];
             }
 
-            public void Remove(object documentOrProjectId)
+            public void Remove(object documentOrProjectId, bool includeProjectState = true)
```

Comment:
Also comment why this parameter exists, and who would want to use it in the non-default state.